canonical-web-and-design/maas-ui
# Fix issues from KVM list design QA.

## Done

- Fixed issues from canonical-web-and-design/MAAS-squad#1955
  - Changed column visibility at different screen sizes
  - Slightly improved mobile styling
  - Built Popover component, and built extended CPUPopover, RamPopover and StoragePopover components from it

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to KVM list and check that the charts show all the down to tablet screen sizes
- Check that the row height remains consistent on all screen sizes
- Check that there is a popover for the RAM and Storage cells

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1955

## Screenshots
### Desktop
![0 0 0 0_8400_MAAS_r_kvm (16)](https://user-images.githubusercontent.com/25733845/88610445-10ad8580-d0ca-11ea-84e3-3fa55b17e833.png)


### Tablet
![2020-07-28_11-58](https://user-images.githubusercontent.com/25733845/88610461-186d2a00-d0ca-11ea-80f3-7653e1e79050.png)


### Mobile
![2020-07-28_11-58_1](https://user-images.githubusercontent.com/25733845/88610468-1b681a80-d0ca-11ea-8355-207ab669d749.png)

### RAM popover
![0 0 0 0_8400_MAAS_r_kvm (19)](https://user-images.githubusercontent.com/25733845/88891717-f8cf3080-d286-11ea-8da3-f4aaec665eb4.png)


### Storage popover
![0 0 0 0_8400_MAAS_r_kvm (20)](https://user-images.githubusercontent.com/25733845/88891732-ff5da800-d286-11ea-9e88-c9aecae11abe.png)


### Storage popover (multiple)
![0 0 0 0_8400_MAAS_r_kvm (21)](https://user-images.githubusercontent.com/25733845/88891743-0389c580-d287-11ea-9b8d-9d1e49b3932f.png)
